### PR TITLE
Refine redis client implementation

### PR DESCRIPTION
**Pull request by NipunaRanasinghe**
## Purpose
This PR is a complete codebase cleanup, which includes the below improvements to the client implementation.
- Improve exception handling (Now the client returns redis module-specific error types, instead of generic `error` values)
- Simplify Java interops in Ballerina client
- Remove deprecated methods usage
- Refactor native utilities and extract duplicated logic into utils
- Simplify Redis connection manager implementation
- Introduce command executor implementation
- Improve documentation and fix compile-time warnings

## Checklist
- [ ] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests
- [ ] Updated the spec
- [ ] Checked native-image compatibility

### CODE REVIEW

File: ballerina/endpoint.bal [hunk 1]
```diff
@@ -23,16 +23,26 @@ import ballerina/jballerina.java;
 public isolated client class Client {
     private ConnectionConfig clientEndpointConfig = {};
```
Comment (ayeshLK) [inline ballerina/endpoint.bal:24]: Is this field used anywhere other than the `init` function ? If not shall we remove this as well :thinking:

File: ballerina/endpoint.bal [hunk 2]
```diff
@@ -23,16 +23,26 @@ import ballerina/jballerina.java;
...
-    final handle datasource;
-
     # Initializes the connector. During initialization you have to pass needed configurations such as host name, port number and password.
     #
     # + config - Configuration for the connector
     # + return - `http:Error` in case of failure to initialize or `null` if successfully initialized
-    public isolated function init(ConnectionConfig config) returns error? {
-        self.datasource = initClient(config);
+    public isolated function init(ConnectionConfig config) returns Error? {
+        Error? initClientResult = self.initClient(self, config);
```
Comment (ayeshLK) [inline ballerina/endpoint.bal:31]: Can't we do this ?
```suggestion
        check self.initClient(self, config);
```

File: ballerina/types.bal [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+# + isClusterConnection - Whether to enable cluster connection or not
+# + ssl - Boolean value depending on whether SSL is enabled or not
+# + startTls - Boolean value depending on whether startTLS is enabled or not
+# + verifyPeer - Boolean value depending on whether peer verification is
+#   enabled or not
+# + database - The database to be used with the connection
+# + connectionTimeout - The timeout value for the connection
+public type Options record {|
+    @display{label: "Client Name"} 
+    string clientName = "";
```
Comment (ayeshLK) [inline ballerina/types.bal:31]: Can't we make this optional ?

File: ballerina/types.bal [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+#
+# + host - The host of the Redis database
+# + password - Password for the database connection
+# + options - Properties for the connection configuration
+@display{label: "Connection Config"} 
+public type ConnectionConfig record {|
+    @display{label: "Host"} 
+    string host = "localhost";
+    @display{label: "Password"} 
+    string password = "";
```
Comment (ayeshLK) [inline ballerina/types.bal:58]: ditto

File: ballerina/endpoint.bal (hunk 1 above)
Comment (NipunaRanasinghe) [reply ballerina/endpoint.bal]: +1 removed with https://github.com/ballerina-platform/module-ballerinax-redis/pull/197/commits/6d826c80794537dc41bc07fc1bf660473cc4c8c5

File: ballerina/endpoint.bal (hunk 2 above)
Comment (NipunaRanasinghe) [reply ballerina/endpoint.bal]: modified with https://github.com/ballerina-platform/module-ballerinax-redis/pull/197/commits/6d826c80794537dc41bc07fc1bf660473cc4c8c5

File: ballerina/types.bal (hunk 3 above)
Comment (NipunaRanasinghe) [reply ballerina/types.bal]: Yeah we can do that as its a non-breaking change. Added with https://github.com/ballerina-platform/module-ballerinax-redis/pull/197/commits/6d826c80794537dc41bc07fc1bf660473cc4c8c5

File: ballerina/types.bal (hunk 4 above)
Comment (NipunaRanasinghe) [reply ballerina/types.bal]: Changed with https://github.com/ballerina-platform/module-ballerinax-redis/commit/6d826c80794537dc41bc07fc1bf660473cc4c8c5

**Review by AzeemMuzammil — DISMISSED** [review]
LGTM

**Review by ayeshLK — APPROVED** [review]

**Pull request merged**